politeiawww/proposals.go

```diff
@@ -392,36 +392,42 @@ func convertWWWPropCreditFromDatabasePropCredit(credit user.ProposalCredit) www.
 	}
 }
 
-// getProp gets the most recent verions of the given proposal from the cache
-// then fills in any missing fields before returning the proposal.
-func (p *politeiawww) getProp(token string) (*www.ProposalRecord, error) {
-	log.Tracef("getProp: %v", token)
-
-	r, err := p.cache.Record(token)
-	if err != nil {
-		return nil, err
-	}
-	pr := convertPropFromCache(*r)
-
+// fillProposalMissingFields populates a ProposalRecord struct with the fields
+// that are not stored in the cache.
+func (p *politeiawww) fillProposalMissingFields(pr *www.ProposalRecord) *www.ProposalRecord {
```

Comment:
I noticed two things that are unrelated to your changes, but since we're touching this code lets go ahead and fix them.

1. This should be fetching the number of comments, not the actual comments.  `getProps` does it correctly.
2. I don't like that this is logging errors instead of returning them.  If something is going wrong, we should fail hard so that the route 500s and we can get notified and fix it rather than just letting the error get logged and us never noticing it.  Let's return the errors so that it fails hard.  `getProps` is doing it correctly.